docs/design/2022-01-04-integer-shard-index.md

```diff
@@ -0,0 +1,423 @@
+# Integer shard index
+
+- RFC PR: https://github.com/tikv/rfcs/pull/0084
+- Tracking Issue: https://github.com/pingcap/tidb/issues/31040
+
+## Summary
+
+Hot index reduces the write scalability of the TiDB cluster when the written data is monotonically increasing. We find out a proposal to solve the problem that is using an expression index to scatter the hot index, the new expression index is called a shard index.
+
+The followers is the scalability of hot index and shard index.
+
+| 测试时间 | 并发数 | tidb数 | tikv数 | tidbCPU | tikvCPU | duration(95)                 | QPS(K) | 扩展性 | 备注   |
+| -------- | ------ | ------ | ------ | ------- | ------- | ---------------------------- | ------ | ------ | ------ |
+| 17 14:30 | 800    | 3      | 3      | 4700    | 3350    | insert:  60.5     update:110 | 49.8   |        | 第一轮 |
+| 17 14:50 | 800    | 3      | 3      | 4400    | 3000    | insert:  61     update:111   | 49     |        | 第二轮 |
+| 17 15:40 | 800    | 3      | 4      | 5000    | 3500    | insert:  57     update:103   | 53     | 19.30% | 第一轮 |
+| 17 9:47  | 800    | 3      | 4      | 5100    | 3600    | insert:  56     update:100   | 54     | 30.60% | 第二轮 |
+
+​																					hot index scalability
+
+
+
+| 测试时间 | 并发数 | tidb数 | tikv数 | tidbCPU | tikvCPU | duration                    | QPS  | 扩展性 | 备注   |
+| -------- | ------ | ------ | ------ | ------- | ------- | --------------------------- | ---- | ------ | ------ |
+| 17 18:50 | 800    | 3      | 3      | 5200    | 3100    | insert: 60     update:110   | 47.5 |        | 第一轮 |
+| 17 19:30 | 800    | 3      | 3      | 5200    | 3000    | insert:  60     update:110  | 47.2 |        | 第二轮 |
+| 17 20:10 | 800    | 3      | 4      | 6200    | 3800    | insert:  56.5     update:88 | 57.5 | 63.20% | 第一轮 |
+| 17 20:40 | 800    | 3      | 4      | 6250    | 3800    | insert:  56.5     update:88 | 58.5 | 71.40% | 第二轮 |
+
+​																					shard index scalability
+
+## Motivation
+
+We made a new built-in function "tidb_shard()" as the expression prefix of shard index. There is only one integer parameter for tidb_shard function. What index is a shard index? It must meet the follower rules.
```

Comment:
This part seems to be the detailed design, we could replace the `Motivation` to `Detailed Design` and remove `Detailed Design` title below.